### ISSUE TITLE
FileZilla Failing to install

### ISSUE THREAD

**Issue by jpallagrosi** [Installomator/Installomator#1390]
ERROR : filezilla : need to provide 'downloadURL'

Tried a few things but couldn't fix it.

**Comment by julienfauryramoin** [comment]
Hi,
There's a pull request that should fix it:
https://github.com/Installomator/Installomator/pull/1386/files

**Comment by jpallagrosi** [comment]
Worked for me. Thank you very much for quick response!

**Comment by aarondavidpolley** [comment]
This is failing on Apple Silicon devices with the PR-ed code. Its having an issue parsing the cpu_arch variable and returns an empty downloadURL

```
aaron.polley@Aarons-MacBook-Pro ~ % sudo /usr/local/Installomator/Installomator.sh filezilla           
Password:
2024-03-22 14:01:15 : REQ   : filezilla : ################## Start Installomator v. 10.5, date 2023-10-15
2024-03-22 14:01:15 : INFO  : filezilla : ################## Version: 10.5
2024-03-22 14:01:15 : INFO  : filezilla : ################## Date: 2023-10-15
2024-03-22 14:01:15 : INFO  : filezilla : ################## filezilla
2024-03-22 14:01:15 : INFO  : filezilla : SwiftDialog is not installed, clear cmd file var
2024-03-22 14:01:18 : ERROR : filezilla : need to provide 'downloadURL'
```

**Comment by aarondavidpolley** [comment]
Ok, issue with the "&" symbol that gets outputted. Without doing a PR to the script, this solves how we are trying to deploy it:

```
/usr/local/Installomator/Installomator.sh $label downloadURL=$(curl -fsL https://filezilla-project.org/download.php\?show_all=1 | grep macos-$(arch) | head -n 1 | awk -F '"' '{print $2}' | sed 's/\&/\\&/')
```

**Comment by aarondavidpolley** [comment]
AND on Intel `arch` returns i386, not x86 so it fails there.

Used this code to get around it

```
if [ $(arch) = "i386" ]
then
	cpu="x86"
else
	cpu="arm64"
fi

# Call Installomator with label
/usr/local/Installomator/Installomator.sh $label downloadURL=$(curl -fsL https://filezilla-project.org/download.php\?show_all=1 | grep macos-$cpu | head -n 1 | awk -F '"' '{print $2}' | sed 's/\&/\\&/')
```

Yes, judge me for assuming `arm64`

**Comment by diBarolo** [comment]
seems broken again.
If I curl the site, there is no download link we can grep.

[filezilla.txt](https://github.com/Installomator/Installomator/files/15419151/filezilla.txt)

**Comment by aarondavidpolley** [comment]
Indeed.  Looks like all "content" is replaced by this when you curl it

`<p>Please visit <a href="https://filezilla-project.org/">https://filezilla-project.org/</a></p>`

But `nscurl` works:

```
if [ $(arch) = "i386" ]
then
	cpu="x86"
else
	cpu="arm64"
fi

# Call Installomator with label
/usr/local/Installomator/Installomator.sh filezilla downloadURL=$(nscurl https://filezilla-project.org/download.php\?show_all=1 | grep macos-$cpu | head -n 1 | awk -F '"' '{print $2}' | sed 's/\&/\\&/')
```

**Comment by aarondavidpolley** [comment]
or use this: #1659

**Comment by diBarolo** [comment]
Thanks a lot !

**Comment by adorador** [comment]
It looks like the solution that aarondavidpolley doesn't appear to be working with the Filezilla server any longer:

2024-07-10 15:40:54 : REQ   : filezilla : ################## Start Installomator v. 10.5, date 2023-10-15
2024-07-10 15:40:54 : INFO  : filezilla : ################## Version: 10.5
2024-07-10 15:40:54 : INFO  : filezilla : ################## Date: 2023-10-15
2024-07-10 15:40:54 : INFO  : filezilla : ################## filezilla
2024-07-10 15:40:56 : INFO  : filezilla : BLOCKING_PROCESS_ACTION=tell_user
2024-07-10 15:40:56 : INFO  : filezilla : NOTIFY=silent
2024-07-10 15:40:56 : INFO  : filezilla : LOGGING=INFO
2024-07-10 15:40:56 : INFO  : filezilla : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2024-07-10 15:40:56 : INFO  : filezilla : Label type: bz2
2024-07-10 15:40:56 : INFO  : filezilla : archiveName: FileZilla.bz2
2024-07-10 15:40:56 : INFO  : filezilla : found packageID org.filezilla-project.filezilla installed, version 3.59.0
2024-07-10 15:40:56 : INFO  : filezilla : appversion: 3.59.0
2024-07-10 15:40:56 : INFO  : filezilla : Latest version of FileZilla is 3.67.1
2024-07-10 15:40:56 : REQ   : filezilla : Downloading https://dl2.cdn.filezilla-project.org/client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855 to FileZilla.bz2
2024-07-10 15:40:57 : ERROR : filezilla : error downloading https://dl2.cdn.filezilla-project.org/client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855
2024-07-10 15:40:57 : ERROR : filezilla : File list: 
2024-07-10 15:40:57 : ERROR : filezilla : File type: FileZilla.bz2: cannot open `FileZilla.bz2' (No such file or directory)
2024-07-10 15:40:57 : INFO  : filezilla : Installomator did not close any apps, so no need to reopen any apps.
2024-07-10 15:40:57 : ERROR : filezilla : ERROR: Error downloading https://dl2.cdn.filezilla-project.org/client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855 error:
* Host dl2.cdn.filezilla-project.org:443 was resolved.
* IPv6: (none)
* IPv4: 78.46.136.26
*   Trying 78.46.136.26:443...
* Connected to dl2.cdn.filezilla-project.org (78.46.136.26) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [334 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [19 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [2601 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [264 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=dl2.cdn.filezilla-project.org
*  start date: Jun 16 07:43:14 2024 GMT
*  expire date: Sep 14 07:43:13 2024 GMT
*  subjectAltName: host "dl2.cdn.filezilla-project.org" matched cert's "dl2.cdn.filezilla-project.org"
*  issuer: C=US; O=Let's Encrypt; CN=R10
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://dl2.cdn.filezilla-project.org/client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: dl2.cdn.filezilla-project.org]
* [HTTP/2] [1] [:path: /client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855]
* [HTTP/2] [1] [user-agent: curl/8.6.0]
* [HTTP/2] [1] [accept: */*]
> GET /client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=vsiYBx6iMwEh7W0axudciQ&x=1720654855 HTTP/2
> Host: dl2.cdn.filezilla-project.org
> User-Agent: curl/8.6.0
> Accept: */*
> 
< HTTP/2 404 
< server: nginx
< date: Wed, 10 Jul 2024 22:40:20 GMT
< content-type: text/html
< content-length: 146
< strict-transport-security: max-age=31536000; includeSubDomains; preload
* The requested URL returned error: 404
* Connection #0 to host dl2.cdn.filezilla-project.org left intact
curl: (22) The requested URL returned error: 404

2024-07-10 15:40:57 : REQ   : filezilla : ################## End Installomator, exit code 2

**Comment by sjurlohne** [comment]
Works for me

2024-07-12 12:38:41 : INFO  : filezilla : ################## filezilla
2024-07-12 12:38:42 : INFO  : filezilla : Reading arguments again:
2024-07-12 12:38:42 : INFO  : filezilla : BLOCKING_PROCESS_ACTION=tell_user
2024-07-12 12:38:42 : INFO  : filezilla : NOTIFY=success
2024-07-12 12:38:42 : INFO  : filezilla : LOGGING=INFO
2024-07-12 12:38:42 : INFO  : filezilla : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2024-07-12 12:38:42 : INFO  : filezilla : Label type: bz2
2024-07-12 12:38:42 : INFO  : filezilla : archiveName: FileZilla.bz2
2024-07-12 12:38:42 : INFO  : filezilla : No version found using packageID org.filezilla-project.filezilla
2024-07-12 12:38:42 : INFO  : filezilla : name: FileZilla, appName: FileZilla.app
2024-07-12 12:38:42.682 mdfind[45115:4524549] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2024-07-12 12:38:42.683 mdfind[45115:4524549] [UserQueryParser] Loading keywords and predicates for locale "en"
2024-07-12 12:38:43.014 mdfind[45115:4524549] Couldn't determine the mapping between prefab keywords and predicates.
2024-07-12 12:38:43 : WARN  : filezilla : No previous app found
2024-07-12 12:38:43 : WARN  : filezilla : could not find FileZilla.app
2024-07-12 12:38:43 : INFO  : filezilla : appversion:
2024-07-12 12:38:43 : INFO  : filezilla : Latest version of FileZilla is 3.67.1
2024-07-12 12:38:43 : INFO  : filezilla : 1. download attempt of 6 attempts.
2024-07-12 12:38:43 : REQ   : filezilla : Downloading https://dl2.cdn.filezilla-project.org/client/FileZilla_3.67.1_macos-arm64.app.tar.bz2?h=10cjP6bxjo4sA7KjZO16DA&x=1720784322 to FileZilla.bz2
2024-07-12 12:38:43 : INFO  : filezilla : 1st download arguments:  -fsL --show-error --retry 5
2024-07-12 12:38:43 : REQ   : filezilla : Installing FileZilla
2024-07-12 12:38:43 : INFO  : filezilla : Unzipping FileZilla.bz2
2024-07-12 12:38:44 : INFO  : filezilla : Verifying: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.4UV5uCa7fe/FileZilla.app
2024-07-12 12:38:45 : INFO  : filezilla : Team ID matching: 5VPGKXL75N (expected: 5VPGKXL75N )
2024-07-12 12:38:45 : INFO  : filezilla : Installing FileZilla version 3.67.1 on versionKey CFBundleShortVersionString.
2024-07-12 12:38:45 : INFO  : filezilla : App has LSMinimumSystemVersion: 10.13.2
2024-07-12 12:38:45 : INFO  : filezilla : Copy /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.4UV5uCa7fe/FileZilla.app to /Applications
2024-07-12 12:38:45 : WARN  : filezilla : Changing owner to sjl
2024-07-12 12:38:45 : INFO  : filezilla : Finishing...
2024-07-12 12:38:48 : INFO  : filezilla : No version found using packageID org.filezilla-project.filezilla
2024-07-12 12:38:48 : INFO  : filezilla : App(s) found: /Applications/FileZilla.app
2024-07-12 12:38:48 : INFO  : filezilla : found app at /Applications/FileZilla.app, version 3.67.1, on versionKey CFBundleShortVersionString
2024-07-12 12:38:48 : REQ   : filezilla : Installed FileZilla, version 3.67.1
2024-07-12 12:38:48 : INFO  : filezilla : notifying
2024-07-12 12:38:49 : INFO  : filezilla : Installomator did not close any apps, so no need to reopen any apps.
2024-07-12 12:38:49 : REQ   : filezilla : All done!
2024-07-12 12:38:49 : REQ   : filezilla : ################## End Installomator, exit code 0

**Comment by timnottom** [comment]
Thanks @aarondavidpolley, works like a charm!

**Comment by acodega** [comment]
Believe this is fixed in Installomator 10.6
